metaflow/click_api.py

```diff
@@ -0,0 +1,416 @@
+import sys
+
+if sys.version_info < (3, 8):
+    raise RuntimeError(
+        """
+        The Metaflow Programmatic API requires 'typeguard', which is not supported for python 3.5, 3.6 and 3.7.
```

Comment:
We can definitely still support 3.7 unless we use some very weird stuff.

Reply:
using typeguard 4.0.1